### PR TITLE
Improve CSV loader error handling

### PR DESCRIPTION
**Pull request by Bartimaeus-**
**Summary**
- Allow PlotJuggler to load CSV files with non-monotonic data with a warning (data is sorted to be monotonic) instead of aborting
- Improvements to error handling dialogs with additional information 

**Handling Non-Monotonic data**
The csv loader would reject csv files that didn't have the data sorted strictly monotonically and would abort loading the file if this was detected. This isn't necessary as the push_back() function used to add the data in the TimeseriesBase class already detects this and inserts new values into the correct place. Since non-monotonic time might be unexpected by the user this change brings up a dialog window where the user can choose to abort or continue with extra information that can be viewed:

image

**Other Improvements**
These are mostly quality of life improvements. I got tired of having to switch between PlotJuggler and my csv file trying to figure out where the issue was.
Examples of the other dialog windows:
image

Failed to parse timestamp:
image

### CODE REVIEW

File: plotjuggler_plugins/DataLoadCSV/dataload_csv.cpp [hunk 1]
```diff
@@ -533,14 +540,31 @@ bool DataLoadCSV::readDataFromFile(FileLoadInfo* info, PlotDataMapRef& plot_data
...
+        QMessageBox msgBox;
+        msgBox.setWindowTitle(tr("Error reading file"));
+        msgBox.setText(tr("The number of values at line %1 is %2,\n"
                              "but the expected number of columns is %3.\n"
                              "Aborting...")
-                         .arg(linecount + 1)
+                         .arg(linecount + 2)
                          .arg(string_items.size())
-                         .arg(column_names.size());
+                         .arg(column_names.size()).arg(linecount+2).arg(t_str));
```
Comment (facontidavide) [inline plotjuggler_plugins/DataLoadCSV/dataload_csv.cpp:550]: Am I wrong or you have 5 arg() here and only 3 are required?

Comment (facontidavide) [inline plotjuggler_plugins/DataLoadCSV/dataload_csv.cpp:550]:
```suggestion
                         .arg(column_names.size());
```

**Review by facontidavide — CHANGES_REQUESTED** [review]

**Comment by Bartimaeus-** [comment]
Missing a parenthesis from that last tweak, I will fix tomorrow

**Comment by facontidavide** [comment]
Thanks a lot

**Pull request merged**